### PR TITLE
Fix: ensure the value in range

### PR DESCRIPTION
**Pull request by Liaojinghui**
There is a bug in the compiler where uint `--` can be less than 0:

```
for(uint i = 6; i<7;i--) ;
```

can run forever until oom.

### CODE REVIEW

**Comment by Liaojinghui** [comment]
there are other cases that also need to be fixed, expecially checked and unchecked not working for `--` and `++`, will keep working later.

**Comment by Liaojinghui** [comment]
@shargon please check this out, this is a bug found by @vang1ong7ang

**Review by shargon — APPROVED** [review]

**Comment by Liaojinghui** [comment]
@shargon all tested, good to go.

File: src/Neo.Compiler.CSharp/MethodConvert.cs [hunk 1]
```diff
@@ -3067,6 +3080,11 @@ private void ConvertElementAccessPreIncrementOrDecrementExpression(SemanticModel
         {
             if (operand.ArgumentList.Arguments.Count != 1)
                 throw new CompilationException(operand.ArgumentList, DiagnosticId.MultidimensionalArray, $"Unsupported array rank: {operand.ArgumentList.Arguments}");
+
+
+            // ITypeSymbol sType = model.GetTypeInfo(expression.Expression).Type!;
+            // ITypeSymbol tType = model.GetTypeInfo(expression.Type).Type!;
+
```
Comment (Liaojinghui) [inline src/Neo.Compiler.CSharp/MethodConvert.cs:3087]:
```suggestion

```

File: tests/Neo.Compiler.CSharp.UnitTests/TestClasses/Contract_Inc_Dec.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        {
+            _property = uint.MinValue;
+            checked
+            {
+                --_property;
+                _property--;
+            }
+            return _property;
+        }
+        public static uint UnitTest_Property_Dec_UnChecked()
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestClasses/Contract_Inc_Dec.cs:42]:
```suggestion

        public static uint UnitTest_Property_Dec_UnChecked()
```

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Inc_Dec.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        [TestMethod]
+        public void Test_Local_Dec_UnChecked()
+        {
+            _engine.Reset();
+            var result = _engine.ExecuteTestCaseStandard("unitTest_Local_Dec_UnChecked");
+            Assert.AreEqual(VMState.HALT, _engine.State);
+            Assert.AreEqual(uint.MaxValue - 1, result.Pop().GetInteger());
+        }
+
+
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Inc_Dec.cs:87]:
```suggestion

```

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Inc_Dec.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        [TestMethod]
+        public void Test_Param_Dec_UnChecked()
+        {
+            _engine.Reset();
+            var result = _engine.ExecuteTestCaseStandard("unitTest_Param_Dec_UnChecked", 0);
+            Assert.AreEqual(VMState.HALT, _engine.State);
+            Assert.AreEqual(uint.MaxValue - 1, result.Pop().GetInteger());
+        }
+
+
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Inc_Dec.cs:122]:
```suggestion

```

**Review by shargon — APPROVED** [review]

**Pull request merged**